### PR TITLE
solution

### PR DESCRIPTION
**Pull request by maksym-kostetskyi**
https://maksym-kostetskyi.github.io/react_add-todo-form

### CODE REVIEW

File: src/Types/ToDo.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { User } from './User';
+
+export interface ToDo {
```
Comment (SanyaBratashchuk) [inline src/Types/ToDo.tsx:3]:
```suggestion
export interface Todo {
```

File: src/components/AddPost/NewPost.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  const [completed, setCompleted] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [user, setUser] = useState({
+    id: 0,
+    name: 'Choose a user',
+    username: '',
+    email: '',
+  });
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userWasSelected, setUserWasSelected] = useState(true);
```
Comment (SanyaBratashchuk) [inline src/components/AddPost/NewPost.tsx:27]:
```suggestion
```
You can just check if user has a value

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]
1. Improve validation since I am able to add todo with just spaces 
image

File: src/components/AddPost/NewPost.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      <div className="field">
+        <label htmlFor="Title">
+          Title:{' '}
+          <input
+            name="Title"
+            type="text"
+            data-cy="titleInput"
+            value={title}
+            onChange={event => handleTitleChange(event.target.value)}
+            required={true}
```
Comment (volodymyr-soltys97) [inline src/components/AddPost/NewPost.tsx:71]:
```suggestion
            required
```

File: src/components/AddPost/NewPost.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            placeholder="Enter a title"
+          />
+        </label>
+
+        {title.trim() === '' && count !== 0 ? (
+          <span className="error">Please enter a title</span>
+        ) : (
+          ''
+        )}
```
Comment (volodymyr-soltys97) [inline src/components/AddPost/NewPost.tsx:80]:
```suggestion
        {!title.trim() && count !== 0 && (
          <span className="error">Please enter a title</span>
        )}
```

File: src/components/AddPost/NewPost.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+              </option>
+            ))}
+          </select>
+        </label>
+
+        {userId === 0 && count !== 0 ? (
+          <span className="error">Please choose a user</span>
+        ) : (
+          ''
+        )}
```
Comment (volodymyr-soltys97) [inline src/components/AddPost/NewPost.tsx:109]: Fix the same as above

File: src/components/TodoInfo/TodoInfo.tsx [hunk 6]
```diff
@@ -1 +1,24 @@
...
+      <h2 className="TodoInfo__title">{todo.title}</h2>
+      <UserInfo user={user} />
+    </article>
+  ) : (
+    <article className="TodoInfo" data-id={todo.id}>
+      <h2 className="TodoInfo__title" data-id={todo.id}>
+        {todo.title}
+      </h2>
+      <UserInfo user={user} />
+    </article>
```
Comment (volodymyr-soltys97) [inline src/components/TodoInfo/TodoInfo.tsx:22]: This logic is similar, so it should be combined

File: src/components/TodoInfo/TodoInfo.tsx [hunk 7]
```diff
@@ -1 +1,24 @@
-export const TodoInfo = () => {};
+import { UserInfo } from '../UserInfo';
+import { Todo } from '../../Types/Todo';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
+  const { completed, user } = todo;
```
Comment (volodymyr-soltys97) [inline src/components/TodoInfo/TodoInfo.tsx:9]:
```suggestion
  const { completed, user, title, id } = todo;
```

File: src/components/TodoInfo/TodoInfo.tsx [hunk 8]
```diff
@@ -1 +1,24 @@
...
+
+  return completed ? (
+    <article className="TodoInfo TodoInfo--completed">
+      <h2 className="TodoInfo__title">{todo.title}</h2>
```
Comment (volodymyr-soltys97) [inline src/components/TodoInfo/TodoInfo.tsx:13]:
```suggestion
      <h2 className="TodoInfo__title">{title}</h2>
```

File: src/components/TodoInfo/TodoInfo.tsx [hunk 9]
```diff
@@ -1 +1,24 @@
...
+      <UserInfo user={user} />
+    </article>
+  ) : (
+    <article className="TodoInfo" data-id={todo.id}>
+      <h2 className="TodoInfo__title" data-id={todo.id}>
```
Comment (volodymyr-soltys97) [inline src/components/TodoInfo/TodoInfo.tsx:18]:
```suggestion
    <article className="TodoInfo" data-id={id}>
      <h2 className="TodoInfo__title" data-id={id}>
```

File: src/components/TodoInfo/TodoInfo.tsx [hunk 10]
```diff
@@ -1 +1,24 @@
...
+        {todo.title}
```
Comment (volodymyr-soltys97) [inline src/components/TodoInfo/TodoInfo.tsx:19]:
```suggestion
        {title}
```

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's make your code better

**Review by VitaliyBondarenko1982 — APPROVED** [review]
Looks good.

**Pull request closed without merging**